pkg/service/publish_results.go

```diff
@@ -145,7 +146,11 @@ func (mw logmw) PublishResults(ctx context.Context, nodeKey string, results []di
 	defer func(begin time.Time) {
 		resJSON, _ := json.Marshal(results)
 		uuid, _ := uuid.FromContext(ctx)
-		mw.logger.Log(
+		logger := level.Debug(mw.logger)
```

Comment:
nit: you could drop an assignment per method with declaring a var, and conditionally assigning

Reply:
How do you mean?

Reply:
```golang
var logger log.logger
if err != nil {
  logger := level.Info(mw.logger)
} else {
  logger := level.Debug(mw.logger)
}
```

Reply:
resolved in slack. I'm going to leave as is and assume the compiler / runtime does the right thing.